riscv/mmu.h

```diff
@@ -134,8 +155,10 @@ class mmu_t
 #endif
 
   // template for functions that store an aligned value to memory
-  #define store_func(type) \
-    void store_##type(reg_t addr, type##_t val) { \
+  #define store_func(type, prefix, xlate_flags) \
+    void prefix##_##type(reg_t addr, type##_t val) { \
+      if (xlate_flags) \
+        flush_tlb(); \
```

Comment:
This is heavyweight, but given that guest loads and stores are much rarer than regular loads and stores, it's good for now.

Reply:
I have avoided changing Spike host TLB implementation in this series.

We need at least "V" bit in TLB tags to distinguish between HS/U-mode TLB entries and VS/VU-mode TLB entries. If this is available then we don't need to flush TLB for guest loads and stores.

Updating host TLB for H-extension and removing these TLB flushes can be done separately.

Reply:
Agreed (if it ever needs to be done at all).